### ISSUE TITLE
How to correctly convert sam-3d-body joint rotations into BVH local rotations?

### ISSUE THREAD

**Issue by t-nakahashi** [facebookresearch/sam-3d-body#94]
I am currently investigating whether sam-3d-body can be used for a
motion-capture–like workflow, and I am trying to generate BVH files
programmatically.

At the moment, I am creating BVH files using the following procedure:

1. From the output of estimator.process_one_image in demo.py,
   I extract data.pred_global_rots
   (the per-joint global rotations produced by the image-based estimation).
   *I understand these to be world-space rotations defined relative to the
   canonical pose.*

2. I obtain the rig offsets and global rotations in the T-pose
   by directly calling the model as follows:

   model = torch.jit.load(model_path, map_location="cpu")
   verts, skel_state = model(
       torch.zeros(1, 45),
       torch.zeros(1, 204),
       torch.zeros(1, 72),
       apply_correctives=False
   )

   joint_coords = skel_state[:, :, :3]
   joint_quats  = skel_state[:, :, 3:7]

3. I extract the parent–child hierarchy of the joints from the model.

4. For both the T-pose and the estimated pose,
   I compute per-joint local rotations by multiplying each joint’s global
   rotation by the inverse of its parent joint’s global rotation.

5. Using these offsets and local rotations, I generate a BVH file.

However, when I import this BVH into Blender (with the default import
settings), the pose is not reproduced correctly.
In particular, the joints from both arms onward deviate significantly from
the expected pose.
At the same time, the output vertices are displayed with the correct shape.

My understanding is that BVH-local rotations can be obtained via:

    global rotation → inverse parent rotation → local rotation

However, I suspect that some important step may be missing, such as:

- The possibility that pred_global_rots does not match the joint-local axis
  conventions assumed by BVH
- Missing compensation equivalent to joint orientation / pre-rotation
  in the T-pose (bind pose)
- The definition of BVH local axes (i.e., the relationship between OFFSET
  vectors and rotation axes)

If anyone has attempted a similar BVH export workflow,
or has detailed knowledge about how joint rotations are defined in
sam-3d-body, I would greatly appreciate any insights or information you
could share.
